src/KafkaFlow/Consumers/MessageConsumer.cs

```diff
@@ -21,27 +21,37 @@ internal class MessageConsumer : IMessageConsumer
 
         public string ConsumerName => this.consumerManager.Consumer.Configuration.ConsumerName;
 
+        public bool IsReadonly => this.consumerManager.Consumer.Configuration.IsReadonly;
+
         public string GroupId => this.consumerManager.Consumer.Configuration.GroupId;
 
         public IReadOnlyList<string> Subscription => this.consumerManager.Consumer.Subscription;
 
-        public IReadOnlyList<TopicPartition> Assignment => this.consumerManager.Consumer.Assignment;
+        public IReadOnlyList<TopicPartition> Assignment => this.consumerManager.Consumer.Assignment ?? new List<TopicPartition>();
+
+        public ConsumerFlowStatus? FlowStatus => this.consumerManager.Consumer.FlowManager?.Status;
 
         public string MemberId => this.consumerManager.Consumer.MemberId;
 
         public string ClientInstanceName => this.consumerManager.Consumer.ClientInstanceName;
 
-        public int WorkerCount => this.consumerManager.Consumer.Configuration.WorkerCount;
+        public int WorkersCount => this.consumerManager.Consumer.Configuration.WorkersCount;
 
-        public ConsumerFlowStatus FlowStatus => this.consumerManager.Consumer.FlowManager.Status;
+        public IReadOnlyList<TopicPartition> PausedPartitions => this.consumerManager.Consumer.FlowManager?.PausedPartitions ?? new List<TopicPartition>();
```

Comment:
I think we can use an IEnumerable here and return Enumerable.Empty() instead of new List()